exporter/exporterhelper/queued_retry_experimental.go

```diff
@@ -120,46 +120,54 @@ func newQueuedRetrySender(id config.ComponentID, signal config.DataType, qCfg Qu
 		onTemporaryFailure: qrs.onTemporaryFailure,
 	}
 
-	if !qCfg.PersistentStorageEnabled {
+	if qCfg.StorageID == nil {
 		qrs.queue = internal.NewBoundedMemoryQueue(qrs.cfg.QueueSize, func(item interface{}) {})
 	}
 	// The Persistent Queue is initialized separately as it needs extra information about the component
 
 	return qrs
 }
 
-func getStorageClient(ctx context.Context, host component.Host, id config.ComponentID, signal config.DataType) (*storage.Client, error) {
-	var storageExtension storage.Extension
-	for _, ext := range host.GetExtensions() {
-		if se, ok := ext.(storage.Extension); ok {
-			if storageExtension != nil {
-				return nil, errMultipleStorageClients
+func (qCfg *QueueSettings) getStorageExtension(extensions map[config.ComponentID]component.Extension) (storage.Extension, error) {
+	if qCfg.StorageID != nil {
```

Comment:
Few suggestions:
1. Usually prefer to check for conditions first and return than indenting the code. `if qCfg.StorageID == nil { return nil, nil } `
2. Better to pass a non pointer `ComponentID` and then no need for the func to be on the QueueSettings. You already check for nil in the caller code.

Reply:
Good point, thx. I also noticed I missed a test for the wrong extension type that was in my original PR, re-added while making these changes.